lang/cs/Org.Apache.REEF.Client/Multi/MultiRuntimeConfigurationBuilder.cs

```diff
@@ -0,0 +1,249 @@
+            {
+                throw new ArgumentException("Unsupported runtime for multiruntime " + runtimeName);
+            }
+
+            this.runtimes.Add(runtimeName);
+            return this;
+        }
+
+        public MultiRuntimeConfigurationBuilder SetDefaultRuntime(RuntimeName runtimeName)
+        {
+            if (!this.supportedRuntimes.Contains(runtimeName))
+            {
+                throw new ArgumentException("Unsupported runtime for multiruntime " + runtimeName);
+            }
+
+            if (this.defaultRuntime.IsPresent())
+            {
+                throw new ArgumentException("Default runtime was already added");
+            }
+
+            this.defaultRuntime = Optional<RuntimeName?>.Of(runtimeName);
+            return this;
+        }
+
+        public MultiRuntimeConfigurationBuilder SetSubmissionRuntime(SubmissionRuntimeName runtimeName)
+        {
+            if (this.submissionRunitme.HasValue)
+            {
+                throw new ArgumentException("Submission runtime was already set");
+            }
+
+            this.submissionRunitme = runtimeName;
+            return this;
+        }
+
+        public MultiRuntimeConfigurationBuilder SetMaxEvaluatorsNumberForLocalRuntime(int maxEvaluatorsNumber)
+        {
+            if (maxEvaluatorsNumber <= 0)
+            {
+                throw new ArgumentException("Max evaluators number should be a positive integer");
+            }
+
+            if (this.maxEvaluatorsNumber.HasValue)
+            {
+                throw new ArgumentException("Max evaluators number was already set");
+            }
+
+            this.maxEvaluatorsNumber = maxEvaluatorsNumber;
+            return this;
+        }
+
+        public MultiRuntimeConfigurationBuilder SetHDInsightUsername(string userName)
+        {
+            this.hdiUsername = userName;
+            return this;
+        }
+
+        public MultiRuntimeConfigurationBuilder SetHDInsightPassword(string pwd)
+        {
+            this.hdiPassword = pwd;
+            return this;
+        }
+
+        public MultiRuntimeConfigurationBuilder SetHDInsightUrl(string hdiUrl)
+        {
+            this.hdiUrl = hdiUrl;
+            return this;
+        }
+
+        public MultiRuntimeConfigurationBuilder SetAzureBlobContainerName(string containerName)
+        {
+            this.jobSubmissionDirectoryPrefix = containerName;
+            return this;
+        }
+
+        public MultiRuntimeConfigurationBuilder SetAzureBlobConnectionString(string connectionString)
+        {
+            this.connectionString = connectionString;
+            return this;
+        }
+
+        public IConfiguration Build()
+        {
+            if (!this.submissionRunitme.HasValue)
+            {
+                throw new ArgumentException("Submission runntime was not set");
+            }
+
+            if (!this.defaultRuntime.IsPresent() && this.runtimes.Count == 1)
```

Comment:
runtimes are the valid runtimes for evaluators. Default runtime specified whoich runtime should be used when request doesn't specify one (for backwards compatibility)